Prog/vacant-lots-montgomery.Rmd

```diff
@@ -0,0 +1,929 @@
+         one_unit_flag = ifelse(dwelling_type == "single-family attached" | 
+                                dwelling_type == "single-family semidetached" | 
+                                dwelling_type == "single-family detached",
+                                1,
+                                0),
+         two_unit_flag = ifelse(dwelling_type == "two family dwellings",
+                                1,
+                                0))
+
+
+```
+
+*For Rockville*
+```{r list codesR}
+zoning_rockville %>% 
+  count(dwelling_type) %>% 
+  knitr::kable()
+
+```
+
+
+There are no dwelling types, so the factor is for all.
+
+### Create factor variable
+All rockville zoning codes are unique, so we do not need a factor. We will automatically select "all", or treat it as dwelling type is NA
+
+**NOTE** This will be VERY jurisdiction specific.
+
+**NOTE** If there are no one or two unit limitations, still make the variable and set them all equal to zero. This allows the functionality of the calculation to be more uniform across counties.
+
+```{r dwelling-factorR}
+
+zoning_rockville1 <- zoning_rockville %>% 
+  mutate(dwelling_type = factor(dwelling_type,
+                                levels = c(
+                                           "all"
+                                           )))
+
+
+```
+
+
+
+*Cleaning and Calculation*
+
+##Clean data for the whole juristdiction
+
+Filter out really large lots that should not be part of the calculation
+Fliter out rockville and gaithersburg
+Count how many vacant lots
+```{r filtering}
+
+jur %>% 
+  filter(lotsize_sf>1000000) %>% 
+count(lotsize_sf, propaddress)
+
+mont <- jur %>% 
+  filter(rockflag != 1, gaitflag!=1, notinmont != 1) %>% 
+  filter(lotsize_sf < 1000000) #filter out unreasonably large lots that mess up the calculation
+
+
+count(mont, rockflag, gaitflag, notinmont)
+
+count(mont, category == "vacant")
+
+```
+
+See how many lots have missing key variables
+
+```{r missing}
+
+mont %>% 
+  mutate(num = NA) %>% 
+  group_by(category_detail) %>% 
+  summarize_at(vars(propaddress, lotsize_sf, zoning, num),
+               ~ sum(is.na(.))) %>% 
+  mutate_at(vars(propaddress, lotsize_sf, zoning),
+            .funs = funs(percent = scales::percent(. / num))) %>% 
+  knitr::kable(caption = "Number and percent of missing values")
+
+```
+## Merge with cleaned data
+
+Now that we have selected the correct records from the zoning code file, we can merge with the cleaned property records data. First, we have to do a bit of cleaning of these codes, including getting rid of instances where there is a tie, and correcting some of the notation. We will create a new variable in order to archive the original `zoning` variable.
+
+```{r bk-clean}
+
+
+montfinal <- mont %>% 
+  mutate(zoning_code = case_when(
+    zoning %in% c("CR1.0","CR0.5", "CR2.0", "CR5.0", "cR3", "CR5", "CR3.0", "CR4.0") ~ "CR",
+    zoning %in% c("RT","RT6", "RT6.0") ~ "RT-6.0",
```

Comment:
Nope, all the other RTs (8, 10, 12 are cleaned and assigned to that code. The general RTs without a number went to 6 because that is the most restrictive zoning code